bsonnumpy/bson/bson-iter.c

```diff
@@ -398,48 +480,18 @@ bson_iter_type (const bson_iter_t *iter) /* IN */
    return bson_iter_type_unsafe (iter);
 }
 
-
-/*
- *--------------------------------------------------------------------------
- *
- * _bson_iter_next_internal --
- *
- *       Internal function to advance @iter to the next field and retrieve
- *       the key and BSON type before error-checking.
- *
- * Return:
- *       true if an element was decoded, else false.
- *
- * Side effects:
- *       @key and @bson_type are set.
- *
- *       If the return value is false:
- *        - @iter is invalidated: @iter->raw is NULLed
- *        - @unsupported is set to true if the bson type is unsupported
- *        - otherwise if the BSON is corrupt, @iter->err_off is nonzero
- *        - otherwise @bson_type is set to BSON_TYPE_EOD
- *
- *--------------------------------------------------------------------------
- */
-
-static bool
-_bson_iter_next_internal (bson_iter_t *iter,   /* INOUT */
-                          const char **key,    /* OUT */
-                          uint32_t *bson_type, /* OUT */
-                          bool *unsupported)   /* OUT */
+bool
+bson_iter_next_with_len (bson_iter_t *iter,
```

Comment:
Can you move `bson_iter_next_with_len` to be defined directly _after_ _bson_iter_next_internal like it was before?

Reply:
Ping.